pandamium_datapack/data/pandamium/functions/misc/global_sleeping.mcfunction

```diff
@@ -1,8 +1,7 @@
-
 scoreboard players set <sleeping_divisor> variable 3
-execute store result score <overworld_player_count> variable if entity @a[nbt={Dimension:"minecraft:overworld"}]
+execute store result score <overworld_player_count> variable if entity @a[x=0]
 scoreboard players set <sleeping_players> variable 0
-execute as @a[nbt={Dimension:"minecraft:overworld"}] if data entity @s SleepingX run scoreboard players add <sleeping_players> variable 1
+execute as @a[x=0,scores={time_since_rest=0}] at @s positioned ~ ~.3 ~ unless entity @s[dx=0] run scoreboard players add <sleeping_players> variable 1
```

Comment:
Could you explain what this line does, bit by bit, please?

Reply:
When a player sleeps, their `time_since_rest` score is constantly 0. However, when a player dies, their score is also set to 0, so we need to use another check.
In the `[dx=0]` selector, dy and dz default to 0, so a virtual 0^3 box is created. This box is then expanded by 1 block in the positive x/y/z directions (see MC-123441), so it goes from (x, y+0.3, z) to (x+1, y+1.3, z+1) with x,y,z being the player's coordinates.
Then, it's checked if the box intersects with the player's hitbox. Sleeping players have a hitbox size of 0.2^3, so it fails for them.

Reply:
(i could also leave out the score check; it's only purpose is a slight performance improvement, and i'm not sure how much it matters)

Reply:
Since the hitbox expansion is a confirmed bug, what happens if it gets fixed? Is it relevant or did you just add it for completeness?

Also, could you change the .3 to 0.3? I know a lot of programming languages allow this, but I really don't like it, a . is really easy to miss compared to a 0 and it just looks weird imo.

Reply:
Please also add a short version of your explanation as a comment above that line.
And since you said the score isn't necessary, I think the `unless entity @s[dx=0]` should be pretty cheap, so it shouldn't be a big difference? In that case I'd prefer to have simpler commands and one less scoreboard objective

Reply:
When the hitbox expansion is fixed, the command should still work. I just mentioned it for completeness.

The score method seems to only be twice as cheap, so I will leave out the score check.